tests/testthat/test-shinytest2-landing_popup.R

```diff
@@ -0,0 +1,172 @@
+  app$get_html(id) %>%
+    rvest::read_html() %>%
+    rvest::html_nodes("button") %>%
+    rvest::html_attr("onclick")
+}
+phash <- function(text) paste0("#", text)
+
+
+testthat::test_that(
+  "e2e: app with customized landing_popup_module creates modal containing specified title, content and buttons",
+  {
+    modal_title <- "Custom Landing Popup Module Title"
+    modal_content_message <- "A welcome message!"
+    modal_content <- tags$b(modal_content_message, style = "color: red;")
+
+    modal_btns <- list(
+      list(text = "Proceed"),
+      list(text = "Read more", onclick = "window.open('http://google.com', '_blank')", id = "read"),
+      list(text = "Reject", onclick = "window.close()", id = "close")
+    )
+    modal_buttons <-
+      tagList(
+        shiny::modalButton(modal_btns[[1]]$text),
+        shiny::actionButton(
+          modal_btns[[2]]$id,
+          label = modal_btns[[2]]$text,
+          onclick = modal_btns[[2]]$onclick
+        ),
+        shiny::actionButton(
+          modal_btns[[3]]$id,
+          label = modal_btns[[3]]$text,
+          onclick = modal_btns[[3]]$onclick
+        )
+      )
+
+    app <- TealAppDriver$new(
+      data = simple_teal_data(),
+      modules = modules(
+        landing_popup_module(
+          title = modal_title,
+          content = modal_content,
+          buttons = modal_buttons
+        ),
+        example_module()
+      )
+    )
+
+    app$wait_for_idle(timeout = default_idle_timeout)
+
+    testthat::expect_equal(
+      app$get_text(".modal-title"),
+      modal_title
+    )
+
+    testthat::expect_equal(
+      trimws(app$get_text(".modal-body")),
+      modal_content_message
+    )
+
+    testthat::expect_equal(
+      app$get_text(".btn-default:nth-child(1)"),
+      modal_btns[[1]]$text
+    )
+
+    testthat::expect_equal(
+      app$get_text(phash(modal_btns[[2]]$id)),
+      modal_btns[[2]]$text
+    )
+
+    testthat::expect_equal(
+      extract_onclick(app, phash(modal_btns[[2]]$id)),
+      modal_btns[[2]]$onclick
+    )
+
+    testthat::expect_equal(
+      app$get_text(phash(modal_btns[[3]]$id)),
+      modal_btns[[3]]$text
+    )
+
+    testthat::expect_equal(
+      extract_onclick(app, phash(modal_btns[[3]]$id)),
+      modal_btns[[3]]$onclick
+    )
+
+    app$stop()
+  }
+)
+
+testthat::test_that("e2e: when customized button in landing_popup_module is clicked, it redirects to a certain page", {
+  onclick_text <- "window.open('http://google.com', '_blank')"
+  app <- TealAppDriver$new(
+    data = simple_teal_data(),
+    modules = modules(
+      landing_popup_module(
+        buttons = actionButton("read", "Read more", onclick = onclick_text)
+      ),
+      example_module()
+    )
+  )
+  app$wait_for_idle(timeout = default_idle_timeout)
+
+  onclick_text_app <- app$get_js("document.getElementById(\"read\").onclick.toString()")
+
+  testthat::expect_true(
+    grepl(onclick_text, onclick_text_app, fixed = TRUE)
+  )
```

Comment:
```suggestion
  testthat::expect_match(onclick_text, onclick_text_app, fixed = TRUE)
```

Reply:
with `app$get_onclick("#read")` no `expect_match` is needed